### PR TITLE
Add the ZED Wrapper and Object Detector

### LINKED ISSUE #2: Add the ZED Wrapper and Object Detector
Add the ZED Wrapper and Object Detector

### PR DESCRIPTION
**Pull request by jbrhm**
## Summary
Closes #2 

What features did you add, bugs did you fix, etc? 
I added the ZED Wrapper and Object Detector

### Did you add documentation to the wiki?
No it is already on the wiki

## How was this code tested? 
This code is almost one for one from the original ROS branch

### Did you test this in sim? 
Yes

### Did you test this on the rover?
No, but I did test on the real life ZED

### Did you add unit tests? 
No

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,7 @@
 cmake_minimum_required(VERSION 3.25)
-project(mrover VERSION 2025.0.0 LANGUAGES C CXX)
+project(mrover VERSION 2025.0.0 LANGUAGES C CXX CUDA)
```
Comment (qhdwight) [inline CMakeLists.txt:2]: Not everyone has CUDA - make sure this works on computers w/o that installed

File: perception/zed_wrapper/zed_wrapper.bridge.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        size_t size = msg->step * msg->height;
+        msg->data.resize(size);
+        checkCudaError(cudaMemcpy(msg->data.data(), bgrGpuPtr, size, cudaMemcpyDeviceToHost));
+    }
+
+	auto fillCameraInfoMessages(sl::CalibrationParameters& calibration, sl::Resolution const& resolution,
+                                sensor_msgs::msg::CameraInfo& leftInfoMsg, sensor_msgs::msg::CameraInfo& rightInfoMsg) -> void {
+
+
+        leftInfoMsg.width = resolution.width;
```
Comment (qhdwight) [inline perception/zed_wrapper/zed_wrapper.bridge.cpp:66]: holy not clanged formatted

File: learning/learning.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#include "pch.hpp"
+
+struct Detection {
+    int classId{};
+    std::string className;
+    float confidence{};
+    cv::Rect box;
+};
+
+class Learning {
```
Comment (qhdwight) [inline learning/learning.hpp:12]: Is this really "learning" if it just does a forward inference pass 🤔

File: CMakeLists.txt (hunk 1 above)
Comment (qhdwight) [reply CMakeLists.txt:2]: I think I used enable_language last year if zed sdk was found

File: CMakeLists.txt [hunk 4]
```diff
@@ -4,6 +4,8 @@ project(mrover VERSION 2025.0.0 LANGUAGES C CXX)
 # Supress ROS CMake warning about Python.
 cmake_policy(SET CMP0148 OLD)
 
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (qhdwight) [inline CMakeLists.txt:7]: Already in the cmake list

**Review by niwhsa9 — COMMENTED** [review]
As promised, I _will_ review any PR that ends up in my inbox.

For everyone's sake we need @qhdwight to teach me how to unsubscribe from the automatic notifications.

**Review by alisonryckman — CHANGES_REQUESTED** [review]

**Pull request merged**